### PR TITLE
source-referralhero

### LINKED ISSUE #46037: [connector-request] ReferralHero
This is a **suggestion** to create a connector for **ReferralHero** using the UI Connector Builder.
The website for the connector is https://referralhero.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by bala-ceg**
## What

This PR adds a new connector for source referralhero (source-referralhero).

Contributor-provided connector description:

airbyte connector developed using airbyte 1.0 UI Connector development
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| lists | uuid | DefaultPaginator | ✅ |  ❌  |
| leaderboard |  | DefaultPaginator | ✅ |  ❌  |
| bonuses |  | DefaultPaginator | ✅ |  ❌  |
| subscribers | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/referralhero.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# referralhero
+airbyte connector developed using airbyte 1.0 UI Connector development
```
Comment (natikgadzhi) [inline docs/integrations/sources/referralhero.md:2]: This should explain what the connector does, what referral hero is and what kind of data is synced, not _how_ it's built.

File: docs/integrations/sources/referralhero.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `api_key` | `string` | API Key.  |  |
+| `uuid` | `string` | uuid.  |  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/referralhero.md:9]: Please check this Loom video why you shouldn't use this config parameter and must implement a parent stream based on lists.

Comment (marcosmarxm) [reply docs/integrations/sources/referralhero.md:9]: [Loom](https://www.loom.com/share/9e5d54e80f76440cb4ebcc65242059e3?sid=5ef0bed9-ae11-40fa-9ca6-a3e19eef432e)

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by bala-ceg** [comment]
@marcosmarxm @natikgadzhi I made the suggested changes, please review

File: docs/integrations/sources/referralhero.md (hunk 1 above)
Comment (bala-ceg) [reply docs/integrations/sources/referralhero.md:2]: Updated the Readme.md thanks

File: docs/integrations/sources/referralhero.md (hunk 2 above)
Comment (bala-ceg) [reply docs/integrations/sources/referralhero.md]: implemented parent stream based on the list, thank you

File: airbyte-integrations/connectors/source-referralhero/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# referralhero
+[Referral Hero](https://referralhero.com) is a tool for creating, managing, and analyzing referral programs to boost customer acquisition and engagement.
+With this connector, you can streamline the transfer of campaign-related data for better integration into your analytics, CRM, or marketing platforms.
+
+Referral Hero Source Connector is a designed to sync referral data between your Referral Hero campaigns and your destination airbyte connectors.
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-referralhero/README.md:5]: Really good description, thank you!

File: docs/integrations/sources/referralhero.md (hunk 1 above)
Comment (natikgadzhi) [reply docs/integrations/sources/referralhero.md:2]: Yeah, but why haven't you updated it here as well? ;-)

File: docs/integrations/sources/referralhero.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `api_key` | `string` | API Key.  |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| lists | uuid | No pagination | ✅ |  ❌  |
```
Comment (natikgadzhi) [inline docs/integrations/sources/referralhero.md:13]: No pagination is a red flag. 

No link to the actual docs to verify if that's intended.

**Review by natikgadzhi — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/referralhero.md (hunk 4 above)
Comment (bala-ceg) [reply docs/integrations/sources/referralhero.md]: I have included the pagination

File: airbyte-integrations/connectors/source-referralhero/README.md (hunk 3 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-referralhero/README.md:5]: Updated thank you

**Review by marcosmarxm — APPROVED** [review]

File: docs/integrations/sources/referralhero.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# referralhero
+[Referral Hero](https://referralhero.com) is a tool for creating, managing, and analyzing referral programs to boost customer acquisition and engagement.
+With this connector, you can streamline the transfer of campaign-related data for better integration into your analytics, CRM, or marketing platforms.
+
+Referral Hero Source Connector is a designed to sync referral data between your Referral Hero campaigns and your destination airbyte connectors.
+
+This directory contains the manifest-only connector for `source-referralhero` airbyte connector developed using airbyte 1.0 UI Connector development
```
Comment (natikgadzhi) [inline docs/integrations/sources/referralhero.md:7]:
```suggestion
```

File: docs/integrations/sources/referralhero.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+[Referral Hero](https://referralhero.com) is a tool for creating, managing, and analyzing referral programs to boost customer acquisition and engagement.
+With this connector, you can streamline the transfer of campaign-related data for better integration into your analytics, CRM, or marketing platforms.
+
+Referral Hero Source Connector is a designed to sync referral data between your Referral Hero campaigns and your destination airbyte connectors.
+
+
+## Generate API Token
+Please follow the instructions in the following [referralhero](https://support.referralhero.com/integrate/rest-api) page to generate the api token
+
+airbyte connector developed using airbyte 1.0 UI Connector development
```
Comment (natikgadzhi) [inline docs/integrations/sources/referralhero.md:11]:
```suggestion
```

**Review by natikgadzhi — APPROVED** [review]

**Comment by natikgadzhi** [comment]
Automerge on. Let's see if Vercel works as expected.

**Pull request merged**